timesketch/api/v1/resources/sigma.py

```diff
@@ -89,4 +95,73 @@ def get(self, rule_uuid):
             abort(
                 HTTP_STATUS_CODE_NOT_FOUND, 'No sigma rule found with this ID.')
 
-        return return_rule
+        meta = {'current_user': current_user.username,
+                'rules_count': len(sigma_rules)}
+        return jsonify({'objects': return_rule, 'meta': meta})
+
+class SigmaByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get a Sigma rule by text."""
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        action = form.get('action', '')
+        
+        if action != 'post':
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Action needs to be "post"')
+        if action == 'post':
+            content = form.get('content')
+            if not content:
+                return abort(
+                    HTTP_STATUS_CODE_BAD_REQUEST,
+                    'Missing values from the request.')
+
+        try:
+            sigma_rule = ts_sigma_lib.get_sigma_rule_by_text(content)
+
+        except ValueError:
+            logger.error('Parsing error',
+                         exc_info=True)
```

Comment:
no need to be in two lines

Reply:
also do we want to be more specific here... `Parsing error` does not tell me much (I know you add in the traceroute) but this is for the sysadmin logs, so perhaps something like `unable to parse a user supplied Sigma rule' or something like that.

Reply:
ack, made it more accurate